torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp

```diff
@@ -1216,6 +1283,22 @@ RegisterOperators reg2({
         "aten::hash.generic(t value) -> int",
         hashValue,
         aliasAnalysisFromSchema()),
+
+#define DEFINE_COMPLEX_MIXED_OP(type_a, type_b)                      \
+  Operator(                                                          \
```

Comment:
OOC, why not using the same macro to define all these different schemas?

Reply:
So when type is `float` then , we need to declare the respective variable `real or img` here as `double` and read the stack variable into the `double` variable. Else if we just try to read the `float` variable directly from stack , then the output differs from eager mode. For example here:
 If we use the macro here `DEFINE_COMPLEX_MIXED_OP(float, float)` and try to call create a complex value. This is how it differs in eager and JIT.
```
complex(4.6, 7.2)

In eager: 4.6 + 7.2j

In JIT: 4.5555559 + 7.111123j
```

Reply:
I don't think that `2` is possible. When we call complex( arg1, arg2) it tries to match the type of args with the `Operator definition`. If we end up passing one more `arg` to the macro, then that would mean complex.* would take 3 args instead of 2.